copy/progress_test.go

```diff
@@ -0,0 +1,112 @@
+package copy
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/containers/image/docker/daemon"
+	"github.com/containers/image/signature"
+	"github.com/containers/image/types"
+	"github.com/docker/docker/reference"
+)
+
+// THIS TEST SHOULD BE SLOW. Read inner comments for details.
+func TestCopyProgress(t *testing.T) {
+	interval := 100 * time.Millisecond
+
+	// we shell out to avoid "client is newer than server" errors from docker
+	// which will happen when importing the client straight from github.
+	// This of course presumes docker is in the path.
+	if out, err := exec.Command("/bin/sh", "-c", "docker pull golang:latest").CombinedOutput(); err != nil {
```

Comment:
(or even better, instead of discarding the output, record it so that we can later verify the copy has happened correctly.)